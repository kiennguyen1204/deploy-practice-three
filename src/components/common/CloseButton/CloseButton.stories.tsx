import type { Meta, StoryObj } from '@storybook/react';

// components
import CloseButton from 'components/common/CloseButton';

const meta: Meta<typeof CloseButton> = {
  component: CloseButton
};

type Story = StoryObj<typeof CloseButton>;

export const Default: Story = {
  args: {
    onClick: () => CloseButton
  }
};

export default meta;
