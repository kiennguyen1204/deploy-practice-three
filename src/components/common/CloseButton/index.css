.modal-close-button {
  background: none;
  border: none;
  font-size: var(--font-size-md);
  cursor: pointer;
}
