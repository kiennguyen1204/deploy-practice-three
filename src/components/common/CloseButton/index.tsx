import { ReactElement } from 'react';

interface Props {
  onClick: () => void;
}

const CloseButton: React.FC<Props> = ({ onClick }): ReactElement => (
  <button
    className='modal-close-button'
    onClick={onClick}
    type='button'
    data-testid='modal-close-button'>
    &times;
  </button>
);

export default CloseButton;
