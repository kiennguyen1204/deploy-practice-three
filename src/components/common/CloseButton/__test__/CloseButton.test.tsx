import { render, screen, fireEvent } from '@testing-library/react';

// components
import CloseButton from 'components/common/CloseButton';

describe('CloseButton', () => {
  it('should call the onClick function when clicked', () => {
    // Arrange
    const onClickMock = jest.fn();
    render(<CloseButton onClick={onClickMock} />);

    // Act
    fireEvent.click(screen.getByTestId('modal-close-button'));

    // Assert
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('should render the close button with proper text', () => {
    // Arrange
    const onClickMock = jest.fn();
    render(<CloseButton onClick={onClickMock} />);

    // Assert
    expect(screen.getByRole('button')).toHaveTextContent('Ã—');
  });

  it('should match the snapshot', () => {
    const onClickMock = jest.fn();
    const { container } = render(<CloseButton onClick={onClickMock} />);

    expect(container).toMatchSnapshot();
  });
});
