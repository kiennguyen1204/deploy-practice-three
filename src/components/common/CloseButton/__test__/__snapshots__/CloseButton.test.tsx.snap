// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton should match the snapshot 1`] = `
<div>
  <button
    class="modal-close-button"
    data-testid="modal-close-button"
    type="button"
  >
    Ã—
  </button>
</div>
`;
